edb/pgsql/metaschema.py

```diff
@@ -2043,8 +2040,8 @@ class StringIndexWithBoundsFunction(dbops.Function):
             "detail"
         )
         FROM (
-            SELECT edgedb._int_or_null(
-                edgedb._normalize_array_index("index", char_length("val"))
+            SELECT (
+                edgedb._normalize_array_index("index", char_length("val"))::int
```

Comment:
with these changes, _normalize_array_index can return `int`, so cast here is not needed